lib/endpoints/build.rb

```diff
@@ -28,7 +28,7 @@ def latest(name)
       # +name+:: Name of the Job
       # +build_id+:: ID of the build
       def test_results(name, build_id)
-        @client.api_request(:get, "/job/#{name}/#{build_id}/testReport")
+        @client.api_request(:get, "/job/#{name}/#{build_id}/testReport", :xml)
```

Comment:
I don't like the idea to force it to be `:xml`. This little change may breaks other tooling build on this library, it was `:json` before and they may depend on that response.

I would prefer to change the function signature to 
```
test_results(name, build_id, type = :json)
```

And use `type` in the `api_request`, that way a user can call it with `:xml` as an argument, but existing tools are not broken because they expect a JSON response.

Reply:
My test reports are uploaded as [JUnit XML](https://stackoverflow.com/a/26661423/2063546), are you saying that Jenkins will automatically convert those to JSON?  If so, what format will they be converted to (given that the XML attributes don't have a comparable feature in JSON)?

Reply:
I don't think it has anything extra, it's just the fact we are using json in a lot of other places so we are using that everywhere where we can instead.